File: artifacts/api-server/src/domain/Instructor.ts
import { Person } from "./Person";

export class Instructor extends Person {
  private courseNames: string[];

  constructor(id: number, name: string, email: string, courseNames: string[] = []) {
    super(id, name, email);
    this.courseNames = courseNames;
  }

  getRole(): string {
    return "instructor";
  }

  getCourses(): string[] {
    return [...this.courseNames];
  }
}

File: server/src/domain/instructor.py
"""Instructor domain model — extends Person."""
from typing import List
from .person import Person


class Instructor(Person):
    def __init__(self, id: int, name: str, email: str, course_names: List[str] = None):
        super().__init__(id, name, email)
        self.course_names: List[str] = course_names or []

    def get_role(self) -> str:
        return "instructor"

    def get_courses(self) -> List[str]:
        return self.course_names
